source/bao_hyp/config.rst

```diff
@@ -65,39 +65,404 @@ describe the image running on the VM (see `Guest Image`_);
+delineated by its virtual address in the second stage. Generally, Bao manages the allocation of
+physical memory for these regions. However, there is an option to define the physical memory region
+where the specific region will be mapped.
+
+For each VM, users can define multiple memory regions. To facilitate this, users first define the
+total number of memory regions via the `region_num` parameter:
+
+- **region_num** [mandatory] - defines the number of memory regions in the VM, specifically, the
+  number of `vm_mem_region` entries in the `vm_platform`'s `regions` list.
+
+Then, each memory region is described by populating the `struct vm_mem_region`:
+
+.. code-block:: c
+
+    struct vm_mem_region {
+        paddr_t base;
+        size_t size;
+        bool place_phys;
+        paddr_t phys;
+    };
+
+where:
+
+- **base** [mandatory] - corresponds to the base virtual address of the memory region;
+- **size**  [mandatory] -  corresponds to the size of the memory region;
+
+.. note::
+  It is mandatory for `base` and `size` to align with the smallest page size of the architecture.
+  For MMU systems, this typically aligns to 4K, while for MPU systems, it aligns to 64 bytes.
+
+- **place_phys** [optional] - the memory region is mapped into the virtual memory, and it's
+  important to note that the virtual address (VA) might not necessarily be the same as the physical
+  address (PA). When "place_phys" is set to true, the virtual address corresponds to the physical
+  address. If ``place_phys`` equals to true, it allows to specify the physical address of the
+  memory region. By default, ``place_phys`` equals to false;
+
+.. note::
+  In MPU systems, place_phys and phys are ignored.
+
+- **phys** [mandatory if ``place_phys`` is true] - it corresponds to the physical address where the
+  memory region should be mapped;
+
+.. note::
+  For enhanced performance, especially in MMU-based targets, it's recommended to align `base` and
+  `size` to the architecture's huge pages (e.g., 2 MiB for Arm and RISC-V). Similarly, if
+  `place_phys` is enabled, aligning `phys` to the architecture's huge pages can also improve
+  performance.
+
+.. note::
+  In MPU systems, place_phys and phys are ignored.
+
+The usage of `place_phys` and `phys` allows users to manually allocate memory and obtain physical
+mappings. This feature provides a means to define the physical memory region explicitly.
 
 3. Inter-Process Communication (IPC)
 ####################################
+.. _Inter-Process Communication (IPC):
+
+Inter-Process Communication (IPC) enables seamless communication between distinct processes in a
+computing system. This mechanism facilitates data exchange, synchronization, and coordination,
+fostering enhanced functionality and system efficiency.
+
+.. figure:: img/ipc.svg
+    :align: center
+    :name: ipc-fig
+
+Bao provides support for IPC, allowing VMs to establish communication channels. The IPC
```

Comment:
To add space below the image and text.
```suggestion
|
```

Reply:
Thanks, fixed!

Reply:
You need to use `|` to produce the empty blank lines. Fix it in other images.